config.go

```diff
@@ -42,3 +71,137 @@ func (c *configuration) GetTagNames() []string {
+				return fmt.Errorf("Health check type is not supported %q for tag %q", config.HealthCheck.Type, tag)
+			}
+			if config.HealthCheck.Path == "" {
+				return fmt.Errorf("No health check path specified for tag %q", tag)
+			}
+
+			if config.BackendService == nil {
+				return fmt.Errorf("No backend service configuration specified for tag %q", tag)
+			}
+			if config.BackendService.Name == "" {
+				return fmt.Errorf("No backend service name specified for tag %q", tag)
+			}
+			if config.BackendService.Affinity != "noaffinity" &&
+				config.BackendService.Affinity != "ipaffinity" &&
+				config.BackendService.Affinity != "ipportaffinity" {
+				return fmt.Errorf("Backend service affinity is not supported %q for tag %q", config.BackendService.Affinity, tag)
+			}
+			if config.BackendService.Host == "" {
+				return fmt.Errorf("No backend service host specified for tag %q", tag)
+			}
+			if config.BackendService.Path == "" {
+				return fmt.Errorf("No backend service path specified for tag %q", tag)
+			}
+			fallthrough
+		case reuseMode:
+			if config.NetworkEndpointGroup == nil {
+				return fmt.Errorf("No network endpoint group configuration specified for tag %q", tag)
+			}
+			if config.NetworkEndpointGroup.Name == "" {
+				return fmt.Errorf("No network endpoint group name specified for tag %q", tag)
+			}
+		default:
+			return fmt.Errorf("Tag %q has invalid mode %q", tag, config.Mode)
+		}
+	}
+	return nil
+}
+
+func parseConfig(configBytes []byte) (*configuration, error) {
+	var c *configuration
+	if err := json.Unmarshal(configBytes, &c); err != nil {
+		return nil, err
+	}
+
+	for tagName, tagConfig := range c.Tags {
+		if tagConfig.Name == "" {
+			tagConfig.Name = tagName
+		}
+		if tagConfig.Name == "" {
+			return nil, fmt.Errorf("Empty tag name")
+		}
+
+		if tagConfig.Mode == tagMode {
+			tagInfo, err := parseTag(tagName)
+			if err != nil {
+				return nil, fmt.Errorf("Failed parsing information for tag %q: %+v", tagName, err)
+			}
+
+			cloudResourceNameRoot := tagInfo.String()
+
+			tagConfig.NetworkEndpointGroup = &networkEndpointGroupConfiguration{
+				Name: makeCloudResourceName("neg", cloudResourceNameRoot),
+			}
+			tagConfig.BackendService = &backendServiceConfiguration{
+				Name:     makeCloudResourceName("bs", cloudResourceNameRoot),
+				CDN:      tagInfo.CDN,
+				Affinity: tagInfo.Affinity,
+				Host:     tagInfo.Host,
+				Path:     tagInfo.Path,
+			}
+			tagConfig.HealthCheck = &healthCheckConfiguration{
+				Name: makeCloudResourceName("hc", cloudResourceNameRoot),
+				Type: tagConfig.HealthCheck.Type,
+				Path: tagConfig.HealthCheck.Path,
+			}
+		}
+	}
+
+	if err := c.validate(); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+// NOTE: Maximum length of resource name is 63 and name must start and end with letter or digit
+func makeCloudResourceName(prefix string, name string) string {
+	n := strings.Join([]string{prefix, name}, "-")
+	if len(n) > maximumNameLength {
```

Comment:
For `maximumNameLength` — By convention, constant names are usually written in uppercase letters. This is for their easy identification and differentiation from variables in the source code.